1.7.0 build fails: error: address of array 'this->storage->audio_otherscene' will always evaluate to 'true'

```
/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.7.0/src/common/dsp/Oscillator.cpp:802:85: error: address of array 'this->storage->audio_otherscene' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
   if( isInSceneB && localcopy[oscdata->p[4].param_id_in_scene].f > 0.f && storage->audio_otherscene )
                                                                        ~~ ~~~~~~~~~^~~~~~~~~~~~~~~~
1 error generated.
```

clang-9
OS: FreeBSD 12.1

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -334,6 +334,7 @@ if( APPLE )
     -Werror=dynamic-class-memaccess
     -Werror=undefined-bool-conversion
     -Werror=bitwise-op-parentheses
+    -Werror=pointer-bool-conversion
     )
   set(OS_INCLUDE_DIRECTORIES
     src/mac

--- a/src/common/dsp/Oscillator.cpp
+++ b/src/common/dsp/Oscillator.cpp
@@ -799,7 +799,7 @@ void osc_audioinput::init_default_values()
 void osc_audioinput::process_block(float pitch, float drift, bool stereo, bool FM, float FMdepth)
 {
    bool useOtherScene = false;
-   if( isInSceneB && localcopy[oscdata->p[4].param_id_in_scene].f > 0.f && storage->audio_otherscene )
+   if( isInSceneB && localcopy[oscdata->p[4].param_id_in_scene].f > 0.f )
    {
       useOtherScene = true;
    }

